Gems/Terrain/Code/Include/Terrain/Passes/TerrainMacroTextureClipmapComputePass.h

```diff
@@ -39,45 +39,47 @@ namespace Terrain
     private:
         TerrainMacroTextureClipmapGenerationPass(const AZ::RPI::PassDescriptor& descriptor);
 
-        void CreateClipmaps();
+        //! Update ClipmapData every frame when view point is changed.
         void UpdateClipmapData();
 
-        static constexpr uint32_t ClipmapStackSize = 4;
+        static constexpr uint32_t ClipmapStackSize = 5;
+        static constexpr uint32_t ClipmapSizeWidth = 1024;
+        static constexpr uint32_t ClipmapSizeHeight = 1024;
 
         struct ClipmapData
         {
-            AZ::Vector3 m_previousViewPosition;
-            AZ::Vector3 m_currentViewPosition;
+            //! The 2D xy-plane view position where the main camera is.
+            //! xy: previous; zw: current.
+            AZ::Vector4 m_viewPosition;
 
-            AZ::Vector3 m_worldBoundsMin;
-            AZ::Vector3 m_worldBoundsMax;
+            // 2D xy-plane world bounds defined by the terrain.
+            // xy: min; zw: max.
+            AZ::Vector4 m_worldBounds;
```

Comment:
Not really safe to use AZ::Vector types in structs that will go directly to the shader. Better to use AZStd::array<float, 4> and then store the data from the Vector using StoreToFloat4().